src/main/java/edu/hawaii/its/api/service/OotbGroupingPropertiesService.java

```diff
@@ -151,13 +171,29 @@ public void setCurrentUser(String currentUser) {
         this.currentUser = currentUser;
     }
 
+    public List<OotbGrouping> getGroupings() {
+        return groupings;
+    }
+
+    public void setGroupings(List<OotbGrouping> groupings) {
+        this.groupings = groupings;
+    }
+
     // Update active user profile with the values from ui request
 
     public OotbActiveProfileResult updateActiveUserProfile(OotbActiveProfile ootbActiveProfile) {
-        List<OotbGrouping> groupings = ootbActiveProfile.getGroupings();
+        // Check the ootb active profile uid that logged in before for caching
+        if (ootbActiveProfileUids.contains(ootbActiveProfile.getUid())) {
+            setCurrentUser(ootbActiveProfile.getUid());
+            return null;
+        }
```

Comment:
It might be better to return back the OotbActiveProfile in a OotbActiveProfileResult rather than null. We should stray away returning null.

Reply:
Sounds good! Thanks